specification/sql/resource-manager/Microsoft.Sql/stable/2018-06-01/managedInstances.json

```diff
@@ -0,0 +1,217 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "2018-06-01",
+    "title": "SqlManagementClient",
+    "description": "The Azure SQL Database management API provides a RESTful set of web APIs that interact with Azure SQL Database services to manage your databases. The API enables users to create, retrieve, update, and delete databases, servers, and other entities."
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Sql/managedInstances/{managedInstanceName}/sqlAgent/current": {
```

Comment:
>sqlAgent [](start = 135, length = 8)

Resource types should be plural. This should be called `sqlAgents` (even if there's always one).

Reply:
You should also have a LIST call to list all the agents. This helps automated tools with discovery (and users who are manually exploring APIs) that don't know that this particular API named their singleton `current`.

---
In reply to: [452574736](https://github.com/Azure/azure-rest-api-specs/pull/8949#discussion_r452574736) [](ancestors = 452574736)

Reply:
You have marked this resolved without responding or making any changes.